GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1828,16 +1819,27 @@ private void Unstage(bool canUseUnstageAll = true)
             }
         }
 
-        [CanBeNull]
-        private GitRevision GetHeadRevision()
+        private (GitRevision workTreeRev, GitRevision indexRev, GitRevision headRev) GetHeadRevisions()
         {
+            GitRevision indexRev;
+            GitRevision headRev;
             var headId = Module.RevParse("HEAD");
             if (headId != null)
             {
-                return new GitRevision(headId);
+                headRev = new GitRevision(headId);
+                indexRev = new GitRevision(ObjectId.IndexId) { ParentIds = new[] { headId } };
+            }
+            else
+            {
+                headRev = null;
+                indexRev = new GitRevision(ObjectId.IndexId);
             }
 
-            return null;
+            var workTreeRev = new GitRevision(ObjectId.WorkTreeId)
+            {
+                ParentIds = new[] { ObjectId.IndexId }
+            };
```

Comment:
Put this on one line as above for `indexRev`, please.

Reply:
squashed